pkgs/development/libraries/glib/default.nix

```diff
@@ -69,9 +69,11 @@ stdenv.mkDerivation rec {
   buildInputs = [ libelf setupHook pcre ]
     ++ optionals stdenv.isLinux [ utillinuxMinimal ]; # for libmount
 
-  nativeBuildInputs = [ pkgconfig perl python gettext ];
+  nativeBuildInputs = [ pkgconfig perl python ];
 
-  propagatedBuildInputs = [ zlib libffi gettext libiconv ];
+  propagatedNativeBuildInputs = [ gettext ];
```

Comment:
This doesn't look right. gettext is definitely a propagated build input. Dependents of glib will use it as a build input, not a native build input, so you won't get the propagated gettext when it's listed native here. I think the old thing was correct. Was there an issue with the previous thing?

Reply:
It might need to be listed in both actually.

Reply:
gettext is both a lib used at runtime and a binary used to process source code, IIRC. So you need the native one for the binary. Not really sure what the deal with propagation is... Is there a test I can use to confirm how propagation of this dep ought to behave?

Reply:
@matthewbauer Well, whatever build failure I experienced in past revisions of nixpkgs is now gone without this change, so I guess I'll remove it. @dezgeg Does that mean this PR no longer needs to target staging?

Reply:
Hmm, yeah, if `gettext` comes with a tool that is used at build-time (I presume it's to parse all the source code of the project to find all the original translations of the strings), then that binary needs to be moved to the `dev` output. (The cross `stdenv` stille needs to be fixed to include such tools in `PATH`, though).

Reply:
At least `gettextize` definitely belongs to `dev`. So that's something to fix at some point.

Yes, no longer need for staging if this glib change is dropped.